bin/build

```diff
@@ -5,13 +5,34 @@ set -e
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
 NS=socrata
 BUILD_DIRS=$@
-BUILD_DIRS=${BUILD_DIRS:-`cd $DIR && ls -d */Dockerfile | cut -f1 -d'/'`}
+# If an image name is provided, validate it can be built
+if [ -n "$BUILD_DIRS" ]; then
+  if [ ! -e "$BUILD_DIRS/Dockerfile" ]; then
+    BUILD_DOCKERFILES=$(find ${BUILD_DIRS} -name Dockerfile)
+    rc=$?; if [[ $rc != 0 ]]; then
+      echo "No Dockerfiles found"
+      exit $rc
+    fi
+    # Find all of the available builds for this image name
+    BUILD_DIRS=$(echo "$BUILD_DOCKERFILES" | sed "s:^\(.*\)/Dockerfile:\1:")
+  fi
+fi
+
+BUILD_DIRS=${BUILD_DIRS:-`cd $DIR && find . -name Dockerfile | sed "s:^./\(.*\)/Dockerfile:\1:"`}
 
 function build_image() {
   local subdir=$1
-  local tag=${subdir}
+  local tag=$(echo $subdir | sed "s:/.*$::")
+  local versions=$(echo $subdir | sed -e "s:^${tag}::" -e "s:/: \
+  :g")
 
-  docker build $BUILD_OPTS -t ${NS}/$tag $DIR/$subdir
+  if [ -n "$versions" ]; then
+    for version in $versions; do
+      docker build $BUILD_OPTS -t ${NS}/$tag:$version $DIR/$subdir
```

Comment:
So just making sure my understanding is correct... and if it is then this looks good.

If you build a new image with the same tag (dockerfile change, security update, etc. etc.) then you have to make sure to explicitly pull that image before building things that depend on it in order to ensure you get the updated version, right?  But once that is done, there is no deploy time problem to deal with.  And none of this really is any different having the version or not having it.